packages/patternfly-4/react-core/src/components/Select/Select.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Select/select.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import SingleSelect from './SingleSelect';
+import SelectToggle from './SelectToggle';
+import { SelectContext } from './selectConstants';
+
+// seed for the aria-labelledby ID
+let currentId = 0;
+
+const propTypes = {
+  /** Content rendered inside the Select */
+  children: PropTypes.node,
+  /** Classes applied to the root of the Select */
+  className: PropTypes.string,
+  /** Flag to indicate if select is expanded */
+  isExpanded: PropTypes.bool,
+  /** Placeholder text of Select */
+  placeholderText: PropTypes.string,
+  /** Array of SelectOption nodes that will be rendered */
+  selectOptions: PropTypes.array,
+  /** Selected item(s) structure */
+  selections: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
+  /** Callback for selection behavior */
+  onSelect: PropTypes.func.isRequired,
+  /** Callback for toggle button behavior */
+  onToggle: PropTypes.func.isRequired,
+  /** Variant of rendered Select */
+  variant: PropTypes.oneOf(['single']),
+  /** Additional props are spread to the container <ul> */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isExpanded: false,
+  selectOptions: null,
+  selections: null,
+  placeholderText: 'Please choose one',
+  variant: 'single'
+};
+
+class Select extends React.Component {
+  componentDidUpdate() {
+    if (!this.props.isExpanded) this.openedOnEnter = false;
+  }
+
+  onEnter = () => {
+    this.openedOnEnter = true;
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      variant,
+      onToggle,
+      onSelect,
+      isExpanded,
+      selectOptions,
+      selections,
+      placeholderText,
+      ...props
+    } = this.props;
+    return (
+      <div
+        className={css(styles.select, isExpanded && styles.modifiers.expanded, className)}
+        ref={ref => {
+          this.parentRef = ref;
```

Comment:
refs should use the new `React.createRef` syntax. https://reactjs.org/docs/refs-and-the-dom.html#creating-refs